tekton-catalog/any-sequencer/cmd/root.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright 2020 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/Knetic/govaluate"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	v1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tektoncdclientset "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"knative.dev/pkg/apis"
+)
+
+var (
+	namespace    string
+	prName       string
+	taskList     string
+	conditions   []string
+	conditionMap map[string][]conditionResult
+)
+
+type conditionResult struct {
+	condition string
+	results   []string
+}
+
+func contains(s []string, str string) bool {
+	for _, v := range s {
+		if v == strings.TrimSpace(str) {
+			return true
+		}
+	}
+	return false
+}
+
+func parse_conditions(condtions []string, tasks *[]string) {
+	conditionMap = make(map[string][]conditionResult)
+	if len(condtions) == 0 {
+		return
+	}
+	// Expect result to have the format of results_taskName_resultName
+	for _, condition := range condtions {
+		operands := regexp.MustCompile(" +").Split(condition, -1)
+		if len(operands) != 3 {
+			log.Printf("The conditon must be as format 'operand1 operator operand2'.")
+			os.Exit(1)
+		}
+		var taskName string
+		var resultName []string
+		resultMatcher := regexp.MustCompile(`results_([^_]*)_([^_]+)`)
+		operand1Results := resultMatcher.FindAllStringSubmatch(operands[0], -1)
+		operand2Results := resultMatcher.FindAllStringSubmatch(operands[2], -1)
+		if len(operand1Results) == 0 && len(operand2Results) == 0 {
+			log.Printf("Must at least contain one result and at most two in one condition for a task.")
+			os.Exit(1)
+		}
+		if len(operand1Results) > 0 && len(operand2Results) > 0 {
+			if operand1Results[0][1] != operand2Results[0][1] {
+				log.Printf("The conditon can only contain results in one task, here's two.")
```

Comment:
shall we ensure this while compiling in DSL? seems that's later in any seq pod.

Reply:
In fact checked it in image, but I agree we need assert on DSL side.